flagz-java/src/main/java/org/flagz/FlagzUnused.java

```diff
@@ -0,0 +1,15 @@
+package org.flagz;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation used on fields of type Flag<?> to indicate that their value is never read (and, as
+ * such, it is useless to specify them). Note that this is subtly stronger than {@link Deprecated}
+ * in that deprecated flags might still be read whereas {@link FlagzUnused} may not.
```

Comment:
@blexim made a good point that flag access potentially crashing might be too high of a price to pay. What do you think about logging an error on access instead? That's quite "loud" as well, even if not quite as strong.